src/policy/mod.rs

```diff
@@ -348,6 +352,7 @@ impl<P> AbstractPolicy<P> {
     /// returns `None`.
     pub fn before_time(mut self, time: u32) -> Option<AbstractPolicy<P>> {
         self = match self {
+            AbstractPolicy::True => AbstractPolicy::True,
```

Comment:
I think there is no need to add a special case here, this is already covered in the last case, 
x => x line 385